src/V3Global.h

```diff
@@ -129,6 +131,12 @@ class V3Global {
     void needTraceDumper(bool flag) { m_needTraceDumper = flag; }
     bool dpi() const { return m_dpi; }
     void dpi(bool flag) { m_dpi = flag; }
+    uint64_t codeSize() const { return m_codeSize; }
+    void codeSizeInc(uint64_t increment) { m_codeSize += increment; }
+    bool useParallelBuild() const {
+        const uint64_t limit = static_cast<uint64_t>(opt.outputSplit());
+        return limit > 0 && codeSize() > limit;
+    }
```

Comment:
I understand the files might be e.g. 4x bigger, though suspect non-linearness is unlikely to matter much between the separate outputs as they don't have much in common. Anyhow I'd really prefer the rule I proposed as it's easily visible from the output if correct or not  basically if we see any __1 file (a split) we know it will parallel.

What I meant is we need a negative test - that is if you have a bug that sets VM_PARALLEL = 1 always, is there some obvious test_regress test that will fail?  (You could do that experiment to find out).  One option would be to copy t_flag_csplit.pl, take away the --output-split settings, and check what test objects get made - should be just one.

Reply:
OK, I'm with you. Please see tests in middle commit. I just pushed.